### PR TITLE
Issue 8495: Only use "noreply" when no sender mail is defined

### PR DESCRIPTION
**Pull request by annando**
This fixes https://github.com/friendica/friendica/issues/8495

### CODE REVIEW

File: src/Util/Emailer.php [hunk 1]
```diff
@@ -64,7 +64,7 @@ public function __construct(IConfig $config, IPConfig $pConfig, App\BaseURL $bas
 		$this->l10n        = $defaultLang;
 
 		$this->siteEmailAddress = $this->config->get('config', 'sender_email');
-		if (empty($sysEmailAddress)) {
+		if (empty($this->siteEmailAddress)) {
```
Comment (Quix0r) [inline src/Util/Emailer.php:67]: Hard to find when PHP is not declarative. Good catch!

File: src/Worker/Delivery.php [hunk 2]
```diff
@@ -564,7 +564,8 @@ private static function deliverMail($cmd, $contact, $owner, $target_item, $thr_p
 				$headers  = 'From: ' . Email::encodeHeader($local_user['username'],'UTF-8') . ' <' . $local_user['email'] . '>' . "\n";
 			}
 		} else {
-			$headers  = 'From: '. Email::encodeHeader($local_user['username'], 'UTF-8') . ' <noreply@' . DI::baseUrl()->getHostname() . '>' . "\n";
+			$sender = DI::config()->get('config', 'sender_email', 'noreply@' . DI::baseUrl()->getHostname());
```
Comment (Quix0r) [inline src/Worker/Delivery.php:567]: Here the local variable `$sender` is being used only once. Is that really wanted?

File: src/Util/Emailer.php (hunk 1 above)
Comment (MrPetovan) [reply src/Util/Emailer.php:67]: What do you mean "not declarative"? Static analyzers would have caught this.

File: src/Worker/Delivery.php (hunk 2 above)
Comment (MrPetovan) [reply src/Worker/Delivery.php:567]: I don't like used once variables as well, but the alternative is to break the below string concatenation on multiple lines which isn't ideal either.

File: src/Util/Emailer.php (hunk 1 above)
Comment (Quix0r) [reply src/Util/Emailer.php:67]: If PHP would be a declarative (paradigm) language, `$sysEmailAddress` cannot be used here.

File: src/Worker/Delivery.php (hunk 2 above)
Comment (Quix0r) [reply src/Worker/Delivery.php:567]: Same here, that is why maybe it needs fixing?

**Pull request merged**